inbound.go

```diff
@@ -336,6 +337,7 @@ func (response *InboundCallResponse) SendSystemError(err error) error {
 	// Fail all future attempts to read fragments
 	response.state = reqResWriterComplete
 	response.systemError = true
+	response.errorType = GetSystemErrorMetricKey(err)
```

Comment:
this is not the only way a response may fail. This is only if the caller specifically wants to return an error.

however, there's errors that happen on the connection (e.g. timeout, cancellation, network failures) which would not go through this path.

We should unit test to make sure that the above scenarios work (timeout and network failure are the ones we tend to care about).